Algorithms/Encoders/NYSIISEncoder.cs

```diff
@@ -0,0 +1,197 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for NYSIIS encoding strings.
+    /// </summary>
+    public class NYSIISEncoder
+    {
+        private static char[] Vowels = new char[] { 'A', 'E', 'I', 'O', 'U' };
+
+        /// <summary>
+        /// Creates a new instance of the NYSIISEncoder.
+        /// </summary>
+        public NYSIISEncoder() {}
+
+        /// <summary>
+        /// Encodes a string using the NYSIIS Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The NYSIIS encoded string (all uppercase)</returns>
+        public string Encode(string text)
+        {
+            text = text.ToUpper();
+            text = StartReplace(text, new (string, string)[] {
+                ("MAC", "MCC"), ("KN", "NN"), ("K", "C"),
+                ("PH", "FF"), ("PF", "FF"), ("SCH", "SSS")
+            });
+            text = EndReplace(text, new (string, string)[] {
+                ("EE", "Yb"), ("IE", "Yb"),
+                ("DT", "Db"), ("RT", "Db"), ("NT", "Db"),
+                ("ND", "Db")
+            });
+
+            string nysiis = $"{text[0]}";
+            for(int i = 1; i < text.Length; i++)
+            {
+                if(Vowels.Contains(text[i]))
+                {
+                    //Vowel: EV -> AF, otherwise ? -> A
+                    text = (i < text.Length - 1 && text[i] == 'E' && text[i + 1] == 'V') ?
+                        text.ReplaceAt(i , "AF", 2) :
+                        text.ReplaceAt(i, "A", 1);
+                }
+                text = RepWith(text, i, new (char, char)[] {
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)